File: backend/app/schemas/market.py
from pydantic import BaseModel


class PriceResponse(BaseModel):
    symbol: str
    name: str | None = None
    price: float
    currency: str
    change: float | None = None
    change_percent: float | None = None
    is_market_open: bool = True
    cached: bool = False


class ExchangeRateResponse(BaseModel):
    pair: str = "USD/KRW"
    rate: float
    change: float | None = None
    change_percent: float | None = None
    cached: bool = False


class MarketTrendItem(BaseModel):
    symbol: str
    name: str
    price: float
    change: float | None = None
    change_percent: float | None = None
    currency: str = "KRW"


class MarketTrendsResponse(BaseModel):
    indices: list[MarketTrendItem]
    gainers: list[MarketTrendItem]
    losers: list[MarketTrendItem]
    cached: bool = False


class MarketSearchResult(BaseModel):
    symbol: str
    name: str
    exchange: str | None = None
    asset_type: str | None = None


class MarketSearchResponse(BaseModel):
    query: str
    results: list[MarketSearchResult]
    cached: bool = False

File: frontend/src/shared/types/market.ts
export interface PriceResponse {
  symbol: string;
  name: string | null;
  price: number;
  currency: string;
  change: number | null;
  change_percent: number | null;
  is_market_open: boolean;
  cached: boolean;
}

export interface ExchangeRateResponse {
  pair: string;
  rate: number;
  change: number | null;
  change_percent: number | null;
  cached: boolean;
}

export interface MarketTrendItem {
  symbol: string;
  name: string;
  price: number;
  change: number | null;
  change_percent: number | null;
  currency: string;
}

export interface MarketTrendsResponse {
  indices: MarketTrendItem[];
  gainers: MarketTrendItem[];
  losers: MarketTrendItem[];
  cached: boolean;
}

export interface MarketSearchResult {
  symbol: string;
  name: string;
  exchange: string | null;
  asset_type: string | null;
}

export interface MarketSearchResponse {
  query: string;
  results: MarketSearchResult[];
  cached: boolean;
}

export interface RefreshPriceRequest { symbol: string; exchange?: string; }
